Assets/MixedRealityToolkit.Services/CameraSystem/MixedRealityCameraSystem.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System.Collections;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.CameraSystem
+{
+    /// <summary>
+    /// The Camera system controls the settings of the main camera.
+    /// </summary>
+    public class MixedRealityCameraSystem : BaseCoreSystem, IMixedRealityCameraSystem
+    {
+        private enum DisplayType
+        {
+            Opaque = 0,
+            Transparent
+        }
+
+        public MixedRealityCameraSystem(
+            IMixedRealityServiceRegistrar registrar,
+            BaseMixedRealityProfile profile = null) : base(registrar, profile)
+        {
+            cameraProfile = (MixedRealityCameraProfile)profile;
+        }
+
+        /// <summary>
+        /// Is the current camera displaying on an Opaque (AR) device or a VR / immersive device
+        /// </summary>
+        public bool IsOpaque
+        {
+            get
+            {
+                currentDisplayType = DisplayType.Opaque;
```

Comment:
it's a little funny that you are changing currentDisplaytype in the getter. What if IsOpaque never gets called? Will currentDisplayType be invalid until IsOpaque is called? Is that a dependency you want to maintain?

Reply:
I'm not sure, to be honest. This was a copy-paste of the existing methods from IMixedRealityToolkit.cs. I wanted to disrupt things as little as possible.